### ISSUE TITLE
post-update.sh to re-clone heimdall and restore backup config

### ISSUE THREAD

**Issue by 1MoreTechGuy** [jsegaert/iocage-plugin-heimdall#4]
I noticed pre-update.sh is taking a backup of the configuration files and sqlite database but doesn't seem to be doing anything with it. I was thinking it should be possible to incorporate a post-update.sh file that could remove the original heimdall directly, clone the latest version, and then restore the files from backup.

Here is a basic version of what I had in mind. Please let me know if I'm missing something or on the wrong path/idea.

```
#!/bin/sh
echo "Removing old Heimdall repo..."
rm -rf /usr/local/www/heimdall
cd /usr/local/www
echo "Downloading latest Heimdall repo..."
git clone https://github.com/linuxserver/Heimdall.git --quiet --depth 1 heimdall
cd heimdall
echo "Copying Heimdall config from backup..."
cp -af /var/backups/heimdall-data/.env ./.env
cp -af /var/backups/heimdall-data/app.sqlite ./database/app.sqlite
cp -afR /var/backups/heimdall-data/public/ ./storage/app/public
chown -R www:www .

echo "Regenerating self-signed SSL certificate..."
/usr/local/bin/heimdall-generate-sslcert.sh > /dev/null 2>&1

echo "Reconfiguring services..."
cp -v /usr/local/etc/heimdall/nginx-heimdall-default.conf /usr/local/etc/nginx/nginx.conf
cp -v /usr/local/etc/php.ini-production /usr/local/etc/php.ini

service php-fpm restart
service nginx restart
```

**Comment by jsegaert** [comment]
You are correct, the intent is to safeguard the configuration files to allow the user to (manually) restore after upgrading the plugin.  I never completely tested the restore procedure.  Note that the official upstream [update procedure](https://github.com/linuxserver/Heimdall#updating) is to "clone the repository or download the zip/tar file with the new version and copy it over the old installation".

**Comment by 1MoreTechGuy** [comment]
Ok Thanks! I ended up 'testing' the restore process in attempt to go from Heimdall 2.2.x to 2.4.x. I tried a few methods for updating the heimdall files (i.e. git pull or copying over existing files) but I ran into an issue with classes missing (see link below). The solution was to delete all the heimdall files, clone clean, and then only copy the three files backed up.

Regardless, I thought I would share in case there was an interest in automating the update of heimdall, but I understand if the desire is to keep it manual. 

https://github.com/linuxserver/Heimdall/issues/781
